src/sensors/SensorManager.cpp

```diff
@@ -41,78 +42,73 @@ namespace SlimeVR
             uint8_t firstIMUAddress = 0;
             uint8_t secondIMUAddress = 0;
 
+            bool sharedIMUAddresses = (PRIMARY_IMU_ADDRESS_ONE == SECONDARY_IMU_ADDRESS_ONE && PRIMARY_IMU_ADDRESS_TWO == SECONDARY_IMU_ADDRESS_TWO);
             {
-#if IMU == IMU_BNO080 || IMU == IMU_BNO085 || IMU == IMU_BNO086
-                firstIMUAddress = I2CSCAN::pickDevice(0x4A, 0x4B, true);
-#elif IMU == IMU_BNO055
-                firstIMUAddress = I2CSCAN::pickDevice(0x29, 0x28, true);
-#elif IMU == IMU_MPU9250 || IMU == IMU_BMI160 || IMU == IMU_MPU6500 || IMU == IMU_MPU6050 || IMU == IMU_ICM20948
-                firstIMUAddress = I2CSCAN::pickDevice(0x68, 0x69, true);
-#else
-#error Unsupported primary IMU
-#endif
+                firstIMUAddress = I2CSCAN::pickDevice(PRIMARY_IMU_ADDRESS_ONE, PRIMARY_IMU_ADDRESS_TWO, true);
+                uint8_t sensorID = 0;
+                if(sharedIMUAddresses && firstIMUAddress != PRIMARY_IMU_ADDRESS_ONE)
```

Comment:
If `firstIMUAddress` is `0`, then it will match this condition, meaning that the `sensorID` will be changed to `1` for an erroneous sensor and could cause confusion about which device is causing the error.

Reply:
In pratice I don't see this being an issue.
firstIMUAddress is 0 when the ic2 scan fails to find any imus.
If the addresses are shared then it means no imus are detected at all.
If the addresses are not shared then sensorID is never set to 1.

Reply:
Thanks for explaining, looks fine then.